src/Unsafe/MutableArray.hs

```diff
@@ -68,29 +68,37 @@ readMutArr mutArr (I# ix) =
   where
     doRead = runRW# $ \stateRW -> readArray# mutArr ix stateRW
 
--- | Grow a mutable array. This is given an array, a given larger size,
--- and it returns a larger array of the given size using the first value
--- to fill in the new cells and copying over all the unchanged cells. This
--- fails for a size smaller than the size of the given array.
-resizeMutArr :: MutArr# a -> Int -> MutArr# a
-resizeMutArr mutArr newSize =
-  let !(# a #) = readMutArr mutArr 0
-  in  resizeMutArrSeed mutArr a newSize
-
--- | Grow a mutable array. This is given an array, a given larger size,
--- and it returns a larger array of the given size using the seed value
--- to fill in the new cells and copying over all the unchanged cells. This
--- fails for a size smaller than the size of the given array.
+-- | Resize a mutable array. That is given an array, a size, and it returns
+-- a new array of the given size using the seed value to fill in the new
+-- cells when necessary and copying over all the unchanged cells.
+--
+-- The size should be positive.
+--
+-- let b = resize a x n,
+--   then length b = n,
+--   and b[i] = a[i] for i < length a,
+--   and b[i] = x for length a <= i < n.
 resizeMutArrSeed :: MutArr# a -> a -> Int -> MutArr# a
 resizeMutArrSeed mutArr x newSize = case newMutArr newSize x of
   newArr -> case copyIntoMutArr mutArr newArr of
     () -> newArr
 
--- | Copy the first mutable array into the second, larger mutable array.
--- This fails if the second array is smaller than the first.
+-- | Same as 'resizeMutArraySeed', but using the first index of the initial
+-- array as the seed value.
+resizeMutArr :: MutArr# a -> Int -> MutArr# a
+resizeMutArr mutArr newSize =
```

Comment:
This means that there is a precondition: the resized array must be non-empty. You should write it explicitly.

Also, all functions with such precondition should have a `HasCallStack` constraint, as I argue in: https://www.tweag.io/blog/2020-04-16-exceptions-in-haskell/ .

Reply:
None of the `Unsafe.MutableArray` functions had `HasCallStack` constraint, although most of them were partial. So I enabled call stacks for the other function too.